### DISCUSSION TITLE
Format output

### DISCUSSION THREAD

**Discussion by burnchar** [MarkPflug/Sylvan#230] [Q&A]
Is there a means to format output generically?
For example, CsvHelper uses class attributes such as `[Format("yyyy-MM-dd")] DateTime MyDate;` or `[Name("Some Header Name")] string CatName;`. If not, is this within the realm of practicality while maintaining a huge performance advantage over CsvHelper?

My context is using the CsvOutputFormatter as part of a web service.

**Comment by MarkPflug** [comment]
For date formatting, you can accomplish this by removing the `AddSylvanCsvFormatters();` call, and instead explicitly register the formatters:
```
OutputFormatters.Add(new CsvOutputFormatter(o => o.DateTimeFormat = "MM/dd/yyyy"));
```
The CsvOutputFormatter accepts a delegate that allow setting the various CSV options supported by the Sylvan CSV library, which includes date formatting.

The header name mapping is not currently supported, but it should be. This would be added via the Sylvan.Data library. Currently, the data binder (which binds DbDataReader to object instances) code in that library uses the `System.Runtime.Serialization.DataMember` to allow mapping to a different column name. The `ObjectDataReader`, which is the inverse of the data binder, should also honor that attribute to allow exposing properties as columns of a different name.

**Comment by MarkPflug** [comment]
Created #231 to track that work.
